components/pthread/pthread.c

```diff
@@ -272,27 +277,40 @@ int pthread_create(pthread_t *thread, const pthread_attr_t *attr,
     task_arg->func = start_routine;
     task_arg->arg = arg;
     pthread->task_arg = task_arg;
-    BaseType_t res = xTaskCreatePinnedToCore(&pthread_task_func,
+    // stack_size is in bytes. This transformation ensures that the units are
+    // transformed to the units used in FreeRTOS.
+    // Note: float division of ceil(m / n) ==
+    //       integer division of (m + n - 1) / n
+    int size_stack = (stack_size + sizeof(StackType_t) - 1) / sizeof(StackType_t);
+    uint32_t memoryLocation = externalmemory ? MALLOC_CAP_SPIRAM : MALLOC_CAP_INTERNAL;
+    StackType_t *stack_for_task = (StackType_t *) heap_caps_calloc(1, size_stack, memoryLocation | MALLOC_CAP_8BIT);
+    if (stack_for_task == NULL) {
+        ESP_LOGE(TAG, "Failed to create task!");
+        free(pthread);
+        free(task_arg);
+    	return ENOMEM;
+    }
+    StaticTask_t *taskTC =  (StaticTask_t *) heap_caps_calloc(1, sizeof(StaticTask_t), MALLOC_CAP_INTERNAL|MALLOC_CAP_8BIT);
+    if (taskTC == NULL) {
+        ESP_LOGE(TAG, "Failed to create task!");
+        free(pthread);
+        free(task_arg);
```

Comment:
Need to add `free(stack_for_task);` here

Reply:
done